AWSClientRuntime/Sources/Auth/AWSCredentialsProviderContainerConfig.swift

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import AwsCommonRuntimeKit
+import ClientRuntime
+
+public struct AWSCredentialsProviderContainerConfig {
+    public let shutDownCallback: ShutDownCallback?
+    public let authToken: String?
+    public let host: String?
+    public let pathAndQuery: String?
```

Comment:
It would be good to order these in the same order as the constructor arguments and assignments

Reply:
updated